src/Sylius/Bundle/UserBundle/DependencyInjection/Configuration.php

```diff
@@ -103,57 +61,102 @@ private function addResourcesSection(ArrayNodeDefinition $node)
-                                                ->scalarNode('registration')->defaultValue(UserRegistrationType::class)->cannotBeEmpty()->end()
+                    ->useAttributeAsKey('name')
+                    ->prototype('array')
+                        ->children()
+                            ->arrayNode('user')
+                                ->addDefaultsIfNotSet()
+                                ->children()
+                                    ->scalarNode('templates')->defaultValue('SyliusUserBundle:User')->end()
+                                    ->variableNode('options')->end()
+                                    ->arrayNode('resetting')
+                                        ->addDefaultsIfNotSet()
+                                        ->children()
+                                            ->arrayNode('token')
+                                                ->addDefaultsIfNotSet()
+                                                ->children()
+                                                    ->scalarNode('ttl')->defaultValue('P1D')->end()
+                                                    ->integerNode('length')
+                                                        ->defaultValue(16)
+                                                        ->min(1)->max(40)
+                                                    ->end()
+                                                    ->scalarNode('field_name')
+                                                        ->defaultValue('passwordResetToken')
+                                                        ->validate()
+                                                        ->ifTrue(function ($tokenFieldName) { return !is_string($tokenFieldName); })
+                                                            ->thenInvalid('Invalid resetting token field "%s"')
+                                                        ->end()
+                                                    ->end()
+                                                ->end()
+                                            ->end()
+                                            ->arrayNode('pin')
+                                                ->addDefaultsIfNotSet()
+                                                ->children()
+                                                    ->integerNode('length')
+                                                        ->defaultValue(4)
+                                                        ->min(1)->max(9)
+                                                    ->end()
+                                                    ->scalarNode('field_name')
+                                                        ->defaultValue('passwordResetToken')
+                                                        ->validate()
+                                                        ->ifTrue(function ($passwordResetToken) { return !is_string($passwordResetToken); })
```

Comment:
Scalar node always has a type `string|null`, so `cannotBeEmpty()` can replace this whole `validate()->ifTrue()->thenInvalid()->end()` chain